include/glow/Optimizer/Optimizer.h

```diff
@@ -29,7 +29,7 @@ enum class CompilationMode {
 };
 
 /// Perform optimizations on the IR representation.
-void optimize(IRFunction &M, CompilationMode mode);
+void optimize(IRFunction &M, CompilationMode mode, Backend *B = nullptr);
```

Comment:
Why is this nullable? Do we ever initialize the execution engine without a backend?

Reply:
This is for the unit tests -- e.g. in IROptTest we call `optimize()` without an execution engine.